index.js

```diff
@@ -155,14 +155,24 @@ Notifications.localNotification = function(details: Object) {
  */
 Notifications.localNotificationSchedule = function(details: Object) {
 	if ( Platform.OS === 'ios' ) {
-		this.handler.scheduleLocalNotification({
+		const iosDetails = {
 			fireDate: details.date.toISOString(),
 			alertBody: details.message,
-			userInfo: details.userInfo
-		});
+			userInfo: details.userInfo,
+			repeatInterval: details.repeatType
+		}
+		// ignore Android only repeatType
+		if (!details.repeatType || details.repeatType === 'time') {
+			delete iosDetails.repeatInterval;
+		}
+		this.handler.scheduleLocalNotification(iosDetails);
 	} else {
 		details.fireDate = details.date.getTime();
 		delete details.date;
+		// ignore iOS only repeatType
+		if (['year', 'month'].includes(details.repeatType)) {
+			delete details.repeatType;
```

Comment:
According to https://github.com/zo0r/react-native-push-notification#local-notifications, 'year' and 'month' are not supported values in Android.

    repeatType: 'day', // (Android only) Repeating interval. Could be one of `week`, `day`, `hour`, `minute, `time`. If specified as time, it should be accompanied by one more parameter 'repeatTime` which should the number of milliseconds between each interval